### ISSUE TITLE
SQL Server user password with characters such as \}{ will break the install

### ISSUE THREAD

**Issue by ukanalyst** [salesagility/SuiteCRM#2554]
#### Issue
Using a password such as this for a SQL Server user will break the installer: 
@^V|\}{

The install page returns the unhelpful error: "The provided database host, username, and/or password is invalid, and a connection to the database could not be established. Please enter a valid host, username and password"

Adding the following code at line 183 of SqlsrvManager.php showed me the correct error in the install log:

installLog(print_r( sqlsrv_errors(), true));

which was:

[message] => An unescaped right brace (}) was found in either the user name or password.  All right braces must be escaped with another right brace (}}).

#### Expected Behavior
Database connected.

#### Actual Behavior
Database didn't connect and vague error displayed.

#### Possible Fix
The password should be escaped in code and ideally clearer logging should be added to return any error codes from the DB driver and deliver these to the install log.

#### Steps to Reproduce

1. Create a MySQL Database
2. Create a user with a password containing the characters @^V|\}{
3. Run the installer
4.

#### Your Environment

* SuiteCRM Version used: 7.7.6
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): n/a
* Environment name and version (e.g. MySQL, PHP 7): PHP 7 and SQL Express 2012
* Operating System and version (e.g Ubuntu 16.04): SQL Server 2012 R2

**Comment by ChangezKhan** [comment]
Hey @ukanalyst, I have used `@` and `{` in my DB user password in SQL Server 2008 R2 Express Edition, but I am able to install CRM without any issues.

**Comment by samus-aran** [comment]
Changing Label from High to Medium as there is a work around i.e. not to use special characters

**Comment by Mac-Rae** [comment]
Confirmed in latest 7.10.x

**Comment by Mephisztoe** [comment]
Confirmed in latest 7.11.15. Used password **U8Ru9%2TDzr<** and spent numerous hours until I found this thread by accident. Changed the sql users' password to something simple for testing purposes and now it works...

**Comment by pgorod** [comment]
@Mephisztoe are you also using SQL Server? I'm wondering if the bug could be more generic.

**Comment by Mephisztoe** [comment]
> @Mephisztoe are you also using SQL Server? I'm wondering if the bug could be more generic.

Yes. Our company is very Microsoft-driven. I am hosting SuiteCRM on a Windows Server 2019 with IIS and MSSQL. Getting file permissions to work and that database provider integrated is not very fun... for my side business, I installed SuiteCRM on a Ubuntu server with Apache and MariaDB. Worked like a charm.
